### PR TITLE
Examples for 7 Functions in Triads.py added solving issue #5498

### LINKED ISSUE #5498: Improving Triads Algorithms Documentation
There are 7 Functions under Triads.

1. [triadic_census](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.triadic_census.html#networkx.algorithms.triads.triadic_census)(G[, nodelist])
2. [random_triad](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.random_triad.html#networkx.algorithms.triads.random_triad)(G)
3. [triads_by_type](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.triads_by_type.html#networkx.algorithms.triads.triads_by_type)(G)
4. [triad_type](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.triad_type.html#networkx.algorithms.triads.triad_type)(G)
5. [is_triad](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.is_triad.html#networkx.algorithms.triads.is_triad)(G)
6. [all_triads](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.all_triads.html#networkx.algorithms.triads.all_triads)(G)
7. [all_triplets](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.all_triplets.html#networkx.algorithms.triads.all_triplets)(G) 

None of these have any examples included. I will like to include some and also improve the documentation on Triads and its Types. 

NetworkX version: 2.6.2

### PR DESCRIPTION
**Pull request by 0ddoes**
Fixes #5498

I have added the examples for 7 Functions in Triads.py file. I manually applied black style to the doc examples as well. I will update the documentation under the same thread. Please let me know if I can improve the examples further.

### CODE REVIEW

File: networkx/algorithms/triads.py
Comment (dschult) [inline networkx/algorithms/triads.py:157]: This might be easier to read (and fit onto the online docs) if you format the printing differently.
```suggestion
    >>> tc = nx.triadic_census(G)
    >>> for key, value in tc.items():
    ...     print(f"{key}: {value},")
    '003': 0,
    '012': 0,
    '102': 0,
    '021D': 0,
    '021U': 0,
    '021C': 0,
    '111D': 0,
    '111U': 0,
    '030T': 2,
    '030C': 2,
    '201': 0,
    '120D': 0,
    '120U': 0,
    '120C': 0,
    '210': 0,
    '300': 0
```
(untested)

**Review by dschult — COMMENTED** [review]
This looks good!  I have a tweak to make the docs fit on a page better.

File: networkx/algorithms/triads.py
Comment (0ddoes) [reply networkx/algorithms/triads.py]: Thanks @dschult for the suggestion! I have edited the file accordingly with the tested and slightly modified variable name.

**Comment by 0ddoes** [comment]
Sir, I am not able to identify why the checks are failing. The last commit I did was simply apply black style but it seems it is failing multiple checks now. Can you please help me out in finding what the issue is? @dschult

**Comment by rossbar** [comment]
The reason tests are failing is likely because the `doctest` format that we use in the docstring examples expects the output to be *exactly* correct, including things like whitespace etc. The failures are currently due to the `triadic_census` example: https://github.com/networkx/networkx/runs/5989988593?check_suite_focus=true.

File: networkx/algorithms/triads.py
Comment (rossbar) [inline networkx/algorithms/triads.py:256]: A very minor consideration, but IMO it's valuable to keep the docstring examples as concise as possible so readers can focus on the important part (i.e. `is_triad` in this case).
```suggestion
    >>> G = nx.DiGraph([(1, 2), (2, 3), (3, 1)])
```

It looks like this comment applies to many of the other examples as well.

File: networkx/algorithms/triads.py
Comment (rossbar) [inline networkx/algorithms/triads.py:259]: A similar thought here - IMO the example would be every so slightly improved by making it as minimal as possible. In this case, adding a single edge is sufficient to break the triad.
```suggestion
    >>> G.add_edge(0, 1)
```

**Review by rossbar — COMMENTED** [review]
Just a couple comments to help tighten up the examples. Ultimately my suggestions boil down to trying to make the example as minimal as possible to illustrate the desired behavior. Removing extraneous information from examples can really make things easier to grok. Thanks for working on these examples!

File: networkx/algorithms/triads.py
Comment (0ddoes) [reply networkx/algorithms/triads.py:256]: Yes Sir! you mentioned this in a previous PR as well. Its my force of habit 😅. I will make sure to edit the others as well. Thanks!

Comment (0ddoes) [reply networkx/algorithms/triads.py:256]: Yes Sir! you mentioned this in a previous PR as well. Its my force of habit 😅. I will make sure to edit the others as well. Thanks!

File: networkx/algorithms/triads.py
Comment (0ddoes) [reply networkx/algorithms/triads.py:258]: Understood Sir! Committing suggestion.

Comment (0ddoes) [reply networkx/algorithms/triads.py:258]: Understood Sir! Committing suggestion.

File: networkx/algorithms/triads.py
Comment (rossbar) [inline networkx/algorithms/triads.py:257]: Just another nit - the code formatting needs to be updated here. Note that this should be done automatically for you if you have the latest [pre-commit hooks installed](https://networkx.org/documentation/latest/developer/contribute.html#development-workflow)

File: networkx/algorithms/triads.py
Comment (rossbar) [inline networkx/algorithms/triads.py:288]: I think it'd be slightly more canonical to write this as:

```suggestion
    >>> list(nx.all_triplets(G))
    [(1, 2, 3), (1, 2, 4), (1, 3, 4), (2, 3, 4)]
```

File: networkx/algorithms/triads.py
Comment (rossbar) [inline networkx/algorithms/triads.py:316]: Instead of creating a list from the generator and showing a few of the instance, we could instead just show all of the triads in the graph by iterating over the generator directly:

```suggestion
    >>> for triad in nx.all_triads(G):
    ...     print(triad.edges)
    [(1, 2), (2, 3), (3, 1)]
    [(1, 2), (4, 1), (4, 2)]
    [(3, 1), (3, 4), (4, 1)]
    [(2, 3), (3, 4), (4, 2)]
```

Just an idea though :) If you prefer to stick with the list approach, then I suggest replacing the list comprehension with the simpler `triads = list(nx.all_triads(G))`

File: networkx/algorithms/triads.py
Comment (0ddoes) [reply networkx/algorithms/triads.py:257]: I think the ```black``` linter is not working for the code in docstring. I had manually changed for the others previously. This was unfortunately left out. I will edit it.

File: networkx/algorithms/triads.py
Comment (0ddoes) [reply networkx/algorithms/triads.py:315]: Yes , thanks I will keep this in mind :)

**Comment by 0ddoes** [comment]
@rossbar  Hey Sir, I have made the edits. Are there ways I should improve it further?

**Comment by rossbar** [comment]
Given the reference to the gallery example proposed in #5528 I'd say we should hold off reviewing this one in too much detail until that's merged!

**Comment by 0ddoes** [comment]
Okay understood!

File: networkx/algorithms/triads.py
Comment (MridulS) [inline networkx/algorithms/triads.py:500]:
```suggestion
    >>> triad = nx.random_triad(G)
    >>> nx.edges(triad)
```

explicit name for the variable.

**Review by MridulS — APPROVED** [review]
LGTM! Just a quick nit pick.

**Pull request merged**
